torch/nn/parallel/distributed.py

```diff
@@ -1883,8 +1947,16 @@ def register_comm_hook(self, state: object, hook: Callable):
             >>> ddp.register_comm_hook(state=None, hook=encode_and_decode)
         """
         self._check_comm_hook(hook)
+        if hook.__name__ in ["bf16_compress_hook", "fp16_compress_hook"]:
+            # If we pass None, then the hook will try to get the world size
+            # by calling `dist.group.WORLD.size()`, which causes compilation
+            # errors. So we pre-decode the process group and pass it to the
+            # hook.
+            if state is None:
```

Comment:
can you check if this is still true? @yifuwang has done some improvements to the funcol API and dynamo tracing support.  even if its not already fixed, wonder if we can just fix the dynamo part and not have to do this workaround.

Reply:
I'm not sure if this is fixed. But I can take a look

Reply:
I just tested, this is not fixed yet. I prefer to land this workaround first and investigate the root cause in a separate PR.